.github/workflows/review.yml

```diff
@@ -43,3 +43,26 @@ jobs:
       run: mv .github/tools/.checkpatch.conf .
     - name: Run checkpatch review
       uses: webispy/checkpatch-action@v9
+  upstream_tests:
+     runs-on: self-hosted
+     timeout-minutes: 120
+     name: upstream tests
+     steps:
+     - uses: actions/checkout@v4
+       with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+     - name: 'Prepare machine'
+       run: |
+          vagrant snapshot restore clean_vm_v1
+          echo "FYI vagrant restore command finished with exit code: $?"
+          sleep 2
+          vagrant status
```

Comment:
cool, FYI that these (sleep and status) aren't "required" but in bringing up mdraid I found them helpful to deal with some flakiness with vagrant every once in a while and its nice in the worfklow output to confirm the status of the box every time you bring it up